src/core/core/rpc_message.cpp

```diff
@@ -325,6 +352,8 @@ void message_ex::seal(bool fill_crc)
 
 bool message_ex::is_right_header() const
 {
+    dassert(hdr_format == NET_HDR_DSN, "only valid for dsn mssage header");
```

Comment:
I don't get it. How about transmitting msgs within rDSN apps but using thrift headers, for instance?